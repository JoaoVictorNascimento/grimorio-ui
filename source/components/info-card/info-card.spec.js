import '../../../internals/test/helper';
import React from 'react';

import InfoCard from './info-card-component';
import Error from '../error';

/** @test {InfoCard} */
describe('InfoCard component', () => {
  /** @test {InfoCard#render} */

  const emptyProps = {
    error: undefined,
    type: undefined,
    tagline: undefined,
    value: undefined,
    onErrorClick: undefined,
    errorMessage: undefined,
    errorBtnText: undefined,
  }

  let wrapper

  beforeAll(() => {
    wrapper = shallow(
      <InfoCard {...emptyProps} />
    );
  })

  describe('when error card', () => {
    const props = {
      onErrorClick: jest.fn()
    }
    const spy = jest.spyOn(props, 'onErrorClick')
    
    it('should render correct', () => {
      wrapper.setProps({
        ...emptyProps,
        error: true,
        onErrorClick: props.onErrorClick,
        errorMessage: 'mensagem de erro',
        errorBtnText: 'botÃ£o do erro'
      })
      expect(wrapper.html()).toMatchSnapshot();
    })

    it('should call onErrorClick when click on button error', () => {

      wrapper.find(Error).props().onErrorClick()
      expect(spy).toHaveBeenCalled();
    })
  })

    it('should render card type money', () => {
      wrapper.setProps({ ...emptyProps, type: 'money', value: 20 })
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('should render card with tagline', () => {
      wrapper.setProps({ ...emptyProps, tagline: 'fevereiro' })
      expect(wrapper.html()).toMatchSnapshot();
    });
});
