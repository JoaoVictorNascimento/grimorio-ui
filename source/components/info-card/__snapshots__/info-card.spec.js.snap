// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoCard component should render card type money 1`] = `"<article open=\\"\\" class=\\"default default shadow\\"><div class=\\"content medium\\"><span class=\\"info\\"><span class=\\"symbol\\">R$</span>20,00</span><span class=\\"tagline\\"></span></div></article>"`;

exports[`InfoCard component should render card with tagline 1`] = `"<article open=\\"\\" class=\\"default default shadow\\"><div class=\\"content medium\\"><span class=\\"info\\"></span><span class=\\"tagline\\"><span>fevereiro</span></span></div></article>"`;

exports[`InfoCard component when error card should render correct 1`] = `"<article open=\\"\\" class=\\"default default shadow\\"><div class=\\"content medium\\"><div class=\\"wrap error\\"><p class=\\"title\\">mensagem de erro</p><button type=\\"button\\" class=\\"btn primary medium\\">bot√£o do erro</button></div></div></article>"`;
